From: Lennart Poettering <lennart@poettering.net>
Date: Fri, 4 Jan 2019 13:24:18 +0100
Subject: json: handle NULL explicitly in json_variant_has_type()

(cherry picked from commit f8c186c9ece5c1c0b89abf52f058efb0ed37e0cb)
(cherry picked from commit 388e534d24cac041bd00b48f27a84d6b31089c67)
---
 src/shared/json.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/shared/json.c b/src/shared/json.c
index 59c4617..c9ee74c 100644
--- a/src/shared/json.c
+++ b/src/shared/json.c
@@ -979,6 +979,8 @@ bool json_variant_has_type(JsonVariant *v, JsonVariantType type) {
         JsonVariantType rt;
 
         v = json_variant_dereference(v);
+        if (!v)
+                return false;
 
         rt = json_variant_type(v);
         if (rt == type)
